tests/e2e/smoke_test.py

```diff
@@ -0,0 +1,284 @@
+# Script that sends http request to local dbt-server for testing purpose.
+# - It doesn't setup the dbt-server and requires user to start dbt-server
+#   locally, the profile should match testing project located
+#   in tests/e2e/fixtures/testing/jaffle_shop(i.e. profile name is jaffle_shop).
+# - It will copy the jaffle_shop project into a temporary directory and test.
+# - Requests are sent one by one, it won't touch concurrency issue.
+# Notice:
+# - Until bug fixed, you need to run dbt deps in tests/e2e/fixtures/jaffle_shop
+#   first, otherwise the initial parse will fail.
+# What can be done in future?
+# - More commands for other use cases.
+# - Switch parse endpoint to async endpoint.
+# - Test other endpoints, e.g sync endpoint.
+#
+# Example run:
+# # Under root directory.
+# python3 -m tests.e2e.smoke_test --local_task_db_path=./working-dir/sql_app.db
```

Comment:
If this is only meant to be run manually (rather than by pytest) does it make sense to let the tester supply the project as well, instead of copying jaffle shop into the fixtures?

Reply:
Offline synced with Rachel and my final plan is that:
- Use jaffle shop which is cloned as store in fixture path as test project. The pros vs using a precompiled msgpack is it's more clear about the project and future proof(we can easily update testcases later). The fixture side is about 1M which is acceptable vs docker image size(200M).
- The smoke test will start and maintain a dbt server. User only need to provide data warehouse credentials in env vars to enable the test. 

Rachel: it seems that this PR is little big large, I'm happy to have a offline review with you if you want!